content/extractor.js

```diff
@@ -16,36 +16,122 @@
 
 // Given DOM element returns array of possible id-links to resolve.
 
+import { flatten } from 'lodash';
+
 export const TAGS_BLACKLIST = [ 'script' ];
 
-export function extractPossibleMatches ($dom) {
-  const tag = $dom.tagName.toLowerCase();
+const EMAIL_PATTERN = /([^\s@]+@[^\s@]+\.[a-z]+)/i;
+const MAILTO_PATTERN = new RegExp(`mailto:${EMAIL_PATTERN.source}`, 'i');
```

Comment:
you can also have something like `mailto:email?subject=123` (but we don't handle those well currently as well)